embedded-graphics/src/primitives/mod.rs

```diff
@@ -1,19 +1,32 @@
 //! Graphics primitives
 
 use crate::geometry::Dimensions;
+use crate::pixelcolor::PixelColor;
 
 pub mod circle;
 pub mod line;
 pub mod rectangle;
+mod styled;
 pub mod triangle;
 
-/// Primitive trait
-pub trait Primitive: Dimensions {}
-
 pub use self::circle::Circle;
 pub use self::line::Line;
 pub use self::rectangle::Rectangle;
+pub use self::styled::Styled;
 pub use self::triangle::Triangle;
+use crate::style::Style;
+
+/// Primitive trait
+pub trait Primitive: Dimensions {
```

Comment:
Can we call this `Stylable` instead? Or at least something as generic. I'm imagining something like a struct in part of a UI library that draws a complex widget (i.e. it's not primitive) which breaks the `Primitive` name.

Reply:
> > Should we also use Styled for fonts? I would prefer to use another struct for styling text, because there was some confusion about what fill_color and stroke_color means for font rendering.
> 
> Yeah that makes sense. Would the fonts also have the same `.into_styled()` method, but which takes a `FontStyle` instead of a `Style`? `.into_styled()` would have to take a generic arg in that case, right?

I haven't really thought about how font styling could be implemented. But I was currently planning to use `Style` and `Styled` exclusively for primitives.
 
> > But this could be simplified by adding a `StyleBuilder` to keep a builder like API:
> Not sure why there's this limitation tbh. Can we remove `WithStyle` now that the only way of styling things is through the `.into_styled()` method on `Primitive`? If so, why not move all the methods on `WithStyle` into `Style`?

Keeping `WithStyle` was temporary to keep font rendering intact.
Having the methods from `WithStyle` implemented on `Style` seemed unnecessary because the fields are public anyway and there is no need for additional setter methods IMO.

> Or, if mutating is preferred, stick a `.clone()` in:
> 
> ```rust
> // ...
> 
> let other_thing = Circle::new(decoration_position, 5)
>     .into_styled(decoration_style.clone().stroke_width(10));
> ```

This only works if `stroke_width` (and the other setters) would take ownership of the `Style` object:
```rust
fn stroke_width(self, width: u32) -> Self { ... ; self }
```
But this wouldn't be as flexible as using a mutable reference for `self`. I think that was the reason I suggested to introduce the `StyleBuilder` instead.

> As a general API convention, I'm starting to think if e-g should mutate by default and require `.clone()` for non-mutations seeing as `.clone()` is going to have the same cost as a copy for all e-g objects (at time of writing). I think you've mentioned this in the past but looking at all the `translate`/`translate_mut` stuff in the diff seems a bit superfluous now.
> 
> Edit: ah yes, in the linked issue no less [#162 (comment)](https://github.com/jamwaffles/embedded-graphics/issues/162#issuecomment-533886275) roll_eyes

At first I also wanted to change the `Transform` implementation in this PR, but there were some problems with this change and I decided to postpone this changes to discuss this further in another issue or PR.

Reply:
Do you think that an UI library is likely to reuse the `Style` struct for styling complex widgets? Most widgets need more styling properties then provided by `Style` and, similar to font styling, would benefit from more specific names.

I was also thinking about changing the names and came up with this:
- Change `primitives` to `shapes`: This might be a better name if the objects are also used for other purposes (#156).
- If `Styled` is only used to apply styling to shapes it could then be called `Primitve`.

Reply:
Yeah my bad, now that I'm a bit more awake the properties on the `Style` struct only really make sense for e-g primitives. 

`shapes`/`Shape` is easier to type but I can't think of a better reason to change the name unless you want to. We're still missing a couple of items like ellipses and polygons/polylines, and `Primitive` and `Shape` work equally well for those cases so up to you if you want to change the name :slightly_smiling_face:

Reply:
If I don't come up with a better reason for this change I'll keep the current names.